### PR TITLE
Switch to Karma for running tests

### PR DESCRIPTION
**Pull request by SirCmpwn**
Advantages:

* Tests are run in an actual browser instead of node. By default it's PhantomJS, which is a headless browser based on Blink or something. Drops the jsdom dependency in favor of the actual DOM
* Source maps are working, so stack traces make sense again
* Coverage reporting is working, this time it is tracking all of our files and it uses a bunch of fancy logic to get more accurate coverage numbers.

Still to do:

* Report coverage to coveralls

### CODE REVIEW

**Comment by SirCmpwn** [comment]
Ready for review+merge. The coverage check is going to fail, but the coverage is only being reduced because I made sure it picks up on every file to test its coverage.

File: karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+var webpack = require("karma-webpack");
```
Comment (eatonphil) [inline karma.conf.js]: If you rename this file to karma.conf.babel.js you'd be able to write this config file in babel.

Comment (SirCmpwn) [inline karma.conf.js]: I don't think that's important.

Comment (eatonphil) [inline karma.conf.js]: Just throwing it out there

**Comment by eatonphil** [comment]
Looks good.

File: .travis.yml [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
-script: |
-    env NODE_ENV=test \
-        node_modules/.bin/babel-node node_modules/.bin/babel-istanbul cover \
-        --report lcovonly \
-        node_modules/.bin/_mocha \
-        --report lcovonly -- -R test/setup.js
-    cat ./coverage/lcov.info | ./node_modules/.bin/coveralls
+script:
+    - |
+        cat <<EOF > src/secrets.js
```
Comment (eatonphil) [inline .travis.yml]: How does this work? Why isn't `> src/secrets.js` after the closing EOF?

Comment (SirCmpwn) [inline .travis.yml]: This is how the bashism works, I dunno why they made it this way.

Comment (eatonphil) [inline .travis.yml]: Would it not work if you moved that part out of the heredoc? I think it would just be clearer

Comment (SirCmpwn) [inline .travis.yml]: No, it doesn't work if you do it after the EOF.

Comment (eatonphil) [inline .travis.yml]: kk

**Comment by eatonphil** [comment]
ready for merge?

**Comment by SirCmpwn** [comment]
Yes.

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**